tests/extras/datasets/pandas/test_generic_dataset.py

```diff
@@ -0,0 +1,381 @@
+        versioned_csv_data_set.save(dummy_dataframe)
+        assert versioned_csv_data_set.exists()
+
+    def test_prevent_overwrite(self, versioned_csv_data_set, dummy_dataframe):
+        """Check the error when attempting to override the data set if the
+        corresponding Generic (csv) file for a given save version already exists."""
+        versioned_csv_data_set.save(dummy_dataframe)
+        pattern = (
+            r"Save path \`.+\` for GenericDataSet\(.+\) must "
+            r"not exist if versioning is enabled\."
+        )
+        with pytest.raises(DataSetError, match=pattern):
+            versioned_csv_data_set.save(dummy_dataframe)
+
+    @pytest.mark.parametrize(
+        "load_version", ["2019-01-01T23.59.59.999Z"], indirect=True
+    )
+    @pytest.mark.parametrize(
+        "save_version", ["2019-01-02T00.00.00.000Z"], indirect=True
+    )
+    def test_save_version_warning(
+        self, versioned_csv_data_set, load_version, save_version, dummy_dataframe
+    ):
+        """Check the warning when saving to the path that differs from
+        the subsequent load path."""
+        pattern = (
+            fr"Save version `{save_version}` did not match load version "
+            fr"`{load_version}` for GenericDataSet\(.+\)"
+        )
+        with pytest.warns(UserWarning, match=pattern):
+            versioned_csv_data_set.save(dummy_dataframe)
+
+    def test_versioning_existing_dataset(
+        self, csv_data_set, versioned_csv_data_set, dummy_dataframe
+    ):
+        """Check the error when attempting to save a versioned dataset on top of an
+        already existing (non-versioned) dataset."""
+        csv_data_set.save(dummy_dataframe)
+        assert csv_data_set.exists()
+        assert csv_data_set._filepath == versioned_csv_data_set._filepath
+        pattern = (
+            f"(?=.*file with the same name already exists in the directory)"
+            f"(?=.*{versioned_csv_data_set._filepath.parent.as_posix()})"
+        )
+        with pytest.raises(DataSetError, match=pattern):
+            versioned_csv_data_set.save(dummy_dataframe)
+
+        # Remove non-versioned dataset and try again
+        Path(csv_data_set._filepath.as_posix()).unlink()
+        versioned_csv_data_set.save(dummy_dataframe)
+        assert versioned_csv_data_set.exists()
+
+
+class TestGenericHtmlDataSet:
+    def test_save_and_load(self, dummy_dataframe, html_data_set):
+        html_data_set.save(dummy_dataframe)
+        df = html_data_set.load()
+        assert_frame_equal(dummy_dataframe, df[0])
+
+
+class TestBadGenericDataSet:
+    def test_bad_file_format_argument(self):
+        ds = GenericDataSet(file_format="kedro", filepath="test.kedro")
+
+        pattern = (
+            "Unable to retrieve `pandas.read_kedro` method, please ensure that your 'file_format' "
+            "parameter has been defined correctly as per the Pandas API https://pandas.pydata.org/docs/reference/io.html"
+        )
```

Comment:
These look like long lines. `black` doesn't re-format strings like this but would be nice to split it across at least 3 lines.